htdocs/compta/facture/class/facture.class.php

```diff
@@ -424,7 +424,7 @@ function create($user,$notrigger=0,$forceduedate=0)
 				    else                                // Old behaviour, if linked_object has only one link per type, so is something like array('contract'=>id1))
 				    {
 				        $origin_id = $tmp_origin_id;
-    					$ret = $this->add_object_linked($origin, $origin_id);
+    					$ret = $this->add_object_linked($this->origin, $origin_id);
```

Comment:
I don't understand this change.
When we create an invoice, we can set the array linked_objects with key 'otherobject", 'idofotherobject', so dolibarr can create link during creation.
So we should have here $origin and $originid coming from the loop on each element of $this->linked_objects
Name of var is not clever. Should be $linkedobjname and $linkedobjid.
I tried a better fix